src/runtime/graph/graph_runtime.cc

```diff
@@ -96,6 +96,44 @@ void GraphRuntime::SetInput(int index, DLTensor* data_in) {
   uint32_t eid = this->entry_id(input_nodes_[index], 0);
   data_entry_[eid].CopyFrom(data_in);
 }
+/*!
+ * \brief set index-th input to the graph without copying the data.
+ * \param index The input index.
+ * \param data_ref The input data that is referred.
+ */
+void GraphRuntime::SetInputZeroCopy(int index, DLTensor* data_ref) {
+  CHECK_LT(static_cast<size_t>(index), input_nodes_.size());
+  uint32_t eid = this->entry_id(input_nodes_[index], 0);
+  const auto& shape = dltensor_entry_shapes_[eid];
+
+  // check the consistency of input shape
+  CHECK_EQ(data_alignment_[eid], GetDataAlignment(*data_ref));
+  CHECK(reinterpret_cast<size_t>(data_ref->data) % kAllocAlignment == 0);
```

Comment:
TVM assumes 64-byte alignment of the memory address. Failing to that will cause subtle coredumps, for example `vmovaps` on unaligned memory. When we setup external input, we need to guard that because we don't know how the memory is allocated.